tests/stress_tests/memcheck_tests/tests.cpp

```diff
@@ -106,3 +106,74 @@ INSTANTIATE_TEST_CASE_P(MemCheckTests, MemCheckTestSuite,
                         ::testing::ValuesIn(
                                 generateTestsParams({"devices", "models"})),
                         getTestCaseName);
+
+TEST_P(MemCheckTestSuite, inference_with_streams) {
+    const auto nstreams = "2";
+    log_info("Inference of InferRequest from network: \"" << model
+                                                          << "\" with precision: \"" << precision
+                                                          << "\" for device: \"" << device << "\""
+                                                          << "\" with strems: \"" << nstreams);
+    auto test_pipeline = [&]{
+        MemCheckPipeline memCheckPipeline;
+
+        Core ie;
+        ie.GetVersions(device);
+        std::map<std::string, std::map<std::string, std::string>> config;
+
+        if (!config.count(device)) 
+            config[device] = {};
+
+        std::map<std::string, std::string>& device_config = config.at(device);
+
+        auto setThroughputStreams = [&] () {
+            const std::string key = device + "_THROUGHPUT_STREAMS";
+            // set to user defined value
+            std::vector<std::string> supported_config_keys = ie.GetMetric(device, METRIC_KEY(SUPPORTED_CONFIG_KEYS));
+            if (std::find(supported_config_keys.begin(), supported_config_keys.end(), key) == supported_config_keys.end()) {
+                throw std::logic_error("Device " + device + " doesn't support config key '" + key + "'! " +
+                                        "Please specify -nstreams for correct devices in format  <dev1>:<nstreams1>,<dev2>:<nstreams2>" +
+                                        " or via configuration file.");
+            }
+            device_config[key] = nstreams;
+        };
+
+    if (device == "CPU") {  
+        // CPU supports few special performance-oriented keys
+        // for CPU execution, more throughput-oriented execution via streams
+        setThroughputStreams();
+    } else if (device == ("GPU")) {
+        // for GPU execution, more throughput-oriented execution via streams
+        setThroughputStreams();
+    } else {
+        throw std::invalid_argument("This device is not supported");
+    }
+
+    for (auto&& item : config) {
+        ie.SetConfig(item.second, item.first);
+    }
+
+    CNNNetwork cnnNetwork = ie.ReadNetwork(model);
+    ExecutableNetwork exeNetwork = ie.LoadNetwork(cnnNetwork, device);
+    InferRequest inferRequest = exeNetwork.CreateInferRequest();
```

Comment:
For scenario which is close to real it is better to create number of inference requests recommended by plugin for set number of streams. Because inference requests itself also affects memory.